lib/service/service.go

```diff
@@ -5299,7 +5299,25 @@ func initSelfSignedHTTPSCert(cfg *servicecfg.Config) (err error) {
 	}
 	cfg.Log.Warningf("Generating self-signed key and cert to %v %v.", keyPath, certPath)
 
-	creds, err := cert.GenerateSelfSignedCert([]string{cfg.Hostname, "localhost"})
+	var hosts = []string{cfg.Hostname, "localhost"}
+
+	// add web public address hosts to self-signed cert
+	for _, addr := range cfg.Proxy.PublicAddrs {
+		proxyHost, _, err := net.SplitHostPort(addr.String())
+		if err != nil {
+			// log and skip error since this is a nice to have
+			cfg.Log.Errorf("Error parsing proxy.public_address %v, skipping adding to self-signed cert: %v", addr.String(), err)
+			continue
+		}
+		// not including IPs included for public addresses for DNS SANS, may want to add to IP SANS in a future version
+		if ip := net.ParseIP(proxyHost); ip != nil {
```

Comment:
I would not do this here, as:

1) it could be considered a breaking change (we will now skip things that were previously included in the cert

2) we might as well just fix things to properly use SANs while we're here

Reply:
I have added a optional IP address option. This will now include those IPs as IP SAN meant to match @strideynet 's earlier comment.